ouroboros-consensus/test-consensus/Test/Consensus/HardFork/Combinator/A.hs

```diff
@@ -0,0 +1,295 @@
+      setTip (Ticked _ st) = st { lgrA_tip = blockPoint blk }
+
+  reapplyLedgerBlock cfg blk st =
+      case runExcept $ applyLedgerBlock cfg blk st of
+        Left  _   -> error "reapplyLedgerBlock: impossible"
+        Right st' -> st'
+
+  ledgerTipPoint = lgrA_tip
+
+instance UpdateLedger BlockA
+
+instance CanForge BlockA where
+  forgeBlock TopLevelConfig{..} _ bno (Ticked sno st) _txs _ = return $ BlkA {
+        blkA_header = HdrA HeaderFields {
+            headerFieldHash     = Lazy.toStrict . B.encode $ unSlotNo sno
+          , headerFieldPrevHash = ledgerTipHash st
+          , headerFieldSlot     = sno
+          , headerFieldNo       = bno
+          }
+      , blkA_body = BodyA (Map.findWithDefault [] sno (lcfgA_forgeTxs ledgerConfig))
+      }
+    where
+      ledgerConfig :: PartialLedgerConfig BlockA
+      ledgerConfig = snd configLedger
+
+instance BlockSupportsProtocol BlockA where
+  validateView _ _ = ()
+
+instance LedgerSupportsProtocol BlockA where
+  protocolLedgerView   _ _ = ()
+  ledgerViewForecastAt _ _ = Just . trivialForecast
+
+instance HasPartialConsensusConfig ProtocolA
+
+instance HasPartialLedgerConfig BlockA where
+  type PartialLedgerConfig BlockA = PartialLedgerConfigA
+
+  completeLedgerConfig _ = (,)
+
+data TxPayloadA = InitiateAtoB
+  deriving (Show, Eq, Generic, NoUnexpectedThunks, Serialise)
+
+instance LedgerSupportsMempool BlockA where
+  data GenTx BlockA = TxA {
+         txA_id      :: TxId (GenTx BlockA)
+       , txA_payload :: TxPayloadA
+       }
+    deriving (Show, Eq, Generic, NoUnexpectedThunks, Serialise)
+
+  type ApplyTxErr BlockA = Void
+
+  applyTx (ei, ledgerConfig) (TxA _ tx) (Ticked sno st) =
+      case tx of
+        InitiateAtoB -> do
+          let tipEpoch       = runIdentity $ epochInfoEpoch ei sno
+              nextEpoch      = succ tipEpoch
+              nextEpochStart = runIdentity $ epochInfoFirst ei nextEpoch
+              distance       = History.countSlots nextEpochStart sno
+          return $ Ticked sno $ st {
+              lgrA_transition = Just (sno, if distance >= k
+                                             then nextEpoch
+                                             else succ nextEpoch)
+            }
+    where
+      SecurityParam k = lcfgA_k ledgerConfig
+
+  reapplyTx = applyTx
+
+  maxTxCapacity _ = maxBound
+  maxTxSize     _ = maxBound
+  txInBlockSize _ = 0
+
+instance HasTxId (GenTx BlockA) where
+  newtype TxId (GenTx BlockA) = TxIdA Int
+    deriving stock   (Show, Eq, Ord, Generic)
+    deriving newtype (NoUnexpectedThunks, Serialise)
+
+  txId = txA_id
+
+instance ShowQuery (Query BlockA) where
+  showResult qry = case qry of {}
+
+instance QueryLedger BlockA where
+  data Query BlockA result
+    deriving (Show)
+
+  answerQuery _ qry = case qry of {}
+  eqQuery qry _qry' = case qry of {}
+
+instance SingleEraBlock BlockA where
+  singleEraParams _   = lcfgA_eraParams
+  singleEraTransition = \_cfg -> fmap snd . lgrA_transition
```

Comment:
We should check that the `SlotNo` in which it was confirmed is sufficiently deep into the chain (i.e., distance from tip `> k`).